services/spar/src/Spar/Scim/User.hs

```diff
@@ -138,18 +139,12 @@ instance Scim.UserDB SparTag Spar where
       brigUser <- getBrigUser' uid
       team' <- getUserTeam' brigUser
       guard $ stiTeam == team'
-      -- TODO: this is a consitency bug. If people change things in brig this
-      -- wont be reflected here. Fix:  Get rid of the scim table and just
-      -- always call synthesizeUser on whatever data we get from brig.
-      -- pure $ synthesizeScimUser brigUser
-      -- [see also](https://github.com/zinfra/backend-issues/issues/1006)
-      getScimUser' uid
+      lift $ synthesizeStoredUser team' brigUser
```

Comment:
this seems quite a change in behaviour  to me. `getUser` should would only return records that are managed by scim; it now instead turns any brig user into a managed user on the fly.

Reply:
We checked: this is the same behavior as on develop, and it also makes sense: you don't want to create a user that already exists because scim is being sneaky about passing it through.

After this PR, we should also be in a good position to start allowing no or (slightly more tricky) more than one IdP while scimming, see https://github.com/zinfra/backend-issues/issues/1377, https://github.com/zinfra/backend-issues/issues/1365.